nixos/roles/k3s/agent.nix

```diff
@@ -0,0 +1,94 @@
+{ config, lib, pkgs, ... }:
+
+with builtins;
+
+let
+  cfg = config.flyingcircus.roles.k3s-agent;
+  fclib = config.fclib;
+  server = fclib.findOneService "k3s-server-server";
+  serverAddress = lib.replaceStrings ["gocept.net"] ["fcio.net"] server.address or "";
+  agentAddress = head fclib.network.srv.v4.addresses;
+  tokenFile = "/var/lib/k3s/secret_token";
+  k3sFlags = [
+    "--flannel-iface=ethsrv"
+    "--node-ip=${agentAddress}"
+    "--data-dir=/var/lib/k3s"
+  ];
+
+in
+{
+  options = {
+    flyingcircus.roles.k3s-agent = {
+      enable = lib.mkEnableOption "Enable K3s (Kubernetes) Agent Node (experimental)";
+      supportsContainers = fclib.mkDisableContainerSupport;
+
+    };
+  };
+
+  config = lib.mkMerge [
+
+    (lib.mkIf cfg.enable {
+
+      assertions = [
+        {
+          assertion = server != null;
+          message = "Invalid Cluster configuration: Kubernetes agent found but no server!";
+        }
+      ];
+
+      boot.supportedFilesystems = [ "nfs" "nfs4" ];
+
+      environment.systemPackages = with pkgs; [
+        config.services.k3s.package
+        bridge-utils
+        nfs-utils
+      ];
+
+      flyingcircus.services.telegraf.inputs = {
+        kubernetes  = [{
+          # Works without auth on localhost.
+          url = "http://localhost:10255";
+          # If the string isn't defined, the kubernetes plugin uses a default location
+          # for the bearer token which we don't use.
+          bearer_token_string = "doesntmatter";
+        }];
+      };
+
+      flyingcircus.activationScripts.kubernetes-apitoken-node = ''
+        mkdir -p /var/lib/k3s
+        umask 077
```

Comment:
please wrap the snippet in a subshell `( )` OR make _our_ activationscripts get wrapped automatically. you could then also refactor the wireguard snippet